From cd216f70dbb302d28fa283a4d7be0daba4082260 Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Wed, 23 Jan 2019 12:01:23 -0800
Subject: [PATCH] MANTA-4035 retire propeller Reviewed by: Kody A Kantor
 <kody@kkantor.com> Approved by: Kody A Kantor <kody@kkantor.com>

---
 Makefile              | 43 -------------------------------------------
 targets.json.in       | 28 ----------------------------
 tools/purge-mg-builds |  1 -
 3 files changed, 72 deletions(-)

diff --git a/Makefile b/Makefile
index b2704c8..35dc0e8 100644
--- a/Makefile
+++ b/Makefile
@@ -2084,49 +2084,6 @@ clean_volapi:
 	(cd build/sdc-volapi && gmake clean)
 
 
-#---- Propeller
-
-_propeller_stamp=$(MANTA_PROPELLER_BRANCH)-$(TIMESTAMP)-g$(MANTA_PROPELLER_SHA)
-PROPELLER_BITS=$(BITS_DIR)/propeller/propeller-pkg-$(_propeller_stamp).tar.bz2
-PROPELLER_IMAGE_BIT=$(BITS_DIR)/propeller/propeller-zfs-$(_propeller_stamp).zfs.gz
-PROPELLER_MANIFEST_BIT=$(BITS_DIR)/propeller/propeller-zfs-$(_propeller_stamp).imgmanifest
-
-.PHONY: propeller
-propeller: $(PROPELLER_BITS) propeller_image
-
-# PATH for propeller build: Ensure /opt/local/bin is first to put gcc 4.5 (from
-# pkgsrc) before other GCCs.
-$(PROPELLER_BITS): build/manta-propeller
-	@echo "# Build propeller: branch $(MANTA_PROPELLER_BRANCH), sha $(MANTA_PROPELLER_SHA), time `date -u +%Y%m%dT%H%M%SZ`"
-	mkdir -p $(BITS_DIR)
-	(cd build/manta-propeller && NPM_CONFIG_CACHE=$(MG_CACHE_DIR)/npm TIMESTAMP=$(TIMESTAMP) BITS_DIR=$(BITS_DIR) gmake release publish)
-	@echo "# Created propeller bits (time `date -u +%Y%m%dT%H%M%SZ`):"
-	@ls -l $(PROPELLER_BITS)
-	@echo ""
-
-.PHONY: propeller_image
-propeller_image: $(PROPELLER_IMAGE_BIT)
-
-$(PROPELLER_IMAGE_BIT): $(PROPELLER_BITS)
-	@echo "# Build propeller_image: branch $(MANTA_PROPELLER_BRANCH), sha $(MANTA_PROPELLER_SHA), time `date -u +%Y%m%dT%H%M%SZ`"
-	./tools/prep_dataset_in_jpc.sh -i "$(PROPELLER_IMAGE_UUID)" -t $(PROPELLER_BITS) \
-		-b "propeller" \
-		-o "$(PROPELLER_IMAGE_BIT)" -p $(PROPELLER_PKGSRC) -O "$(MG_OUT_PATH)" \
-		-t $(PROPELLER_EXTRA_TARBALLS) -n $(PROPELLER_IMAGE_NAME) \
-		-v $(_propeller_stamp) -d $(PROPELLER_IMAGE_DESCRIPTION)
-	@echo "# Created propeller image (time `date -u +%Y%m%dT%H%M%SZ`):"
-	@ls -l $$(dirname $(PROPELLER_IMAGE_BIT))
-	@echo ""
-
-propeller_publish_image: $(PROPELLER_IMAGE_BIT)
-	@echo "# Publish propeller image to SDC Updates repo."
-	$(UPDATES_IMGADM) import -ddd -m $(PROPELLER_MANIFEST_BIT) -f $(PROPELLER_IMAGE_BIT)
-
-clean_propeller:
-	$(RM) -rf $(BITS_DIR)/propeller
-	(cd build/manta-propeller && gmake distclean)
-
-
 #---- Moray
 
 _moray_stamp=$(MORAY_BRANCH)-$(TIMESTAMP)-g$(MORAY_SHA)
diff --git a/targets.json.in b/targets.json.in
index 880ad0c..ebb4306 100644
--- a/targets.json.in
+++ b/targets.json.in
@@ -1301,34 +1301,6 @@ cat <<EOF
     ]
   },
 
-  "propeller": {
-    "appliance": "true",
-    "build_platform": "20151126T062538Z",
-    "image_name": "manta-propeller",
-    "image_description": "Manta propeller",
-    "image_uuid": "fd2cc906-8938-11e3-beab-4359c665ac99",
-    "pkgsrc": [
-      "postgresql91-client-9.1.2",
-      "redis-2.4.13",
-      "zookeeper-client-3.4.3"
-    ],
-    "repos": [
-      {"url": "$MG_GIT_PREFIX/manta-propeller.git"}
-    ],
-    "public": false,
-    "deps": [
-      "https://download.joyent.com/pub/build/sdcnode",
-      "registrar",
-      "amon",
-      "config-agent"
-    ],
-    "tarballs": [
-      {"tarball": "registrar/registrar-pkg-*.tar.gz", "sysroot": "/"},
-      {"tarball": "amon/amon-agent-*.tgz", "sysroot": "/opt"},
-      {"tarball": "config-agent/config-agent-*.tar.gz", "sysroot": "/opt/smartdc"}
-    ]
-  },
-
   "sdcadm": {
     "build_platform": "20151126T062538Z",
     "repos": [
diff --git a/tools/purge-mg-builds b/tools/purge-mg-builds
index 02d5ccf..d2b13a1 100755
--- a/tools/purge-mg-builds
+++ b/tools/purge-mg-builds
@@ -136,7 +136,6 @@ TTL_DAYS_FROM_NAME='{
     "nfsserver": 365,
     "papi": 365,
     "portolan": 365,
-    "propeller": 365,
     "provisioner": 365,
     "rabbitmq": 365,
     "redis": 365,
-- 
2.21.0

